mlir/include/mlir/Analysis/AffineStructures.h

```diff
@@ -887,6 +902,67 @@ class FlatAffineValueConstraints : public FlatAffineConstraints {
   SmallVector<Optional<Value>, 8> values;
 };
 
+/// A FlatAffineRelation represents a set of ordered pairs (domain -> range)
+/// where "domain" and "range" are tuples of identifiers.
```

Comment:
Add that they are actually represented as [domainDims, rangeDims, SymbolDims, LocalDims, Constant]